relay-common/src/constants.rs

```diff
@@ -665,9 +665,70 @@ impl schemars::JsonSchema for MetricUnit {
     }
 }
 
+/// An error parsing a [`MetricName`] or one of its variants.
+#[derive(Clone, Copy, Debug)]
+pub struct ParseMetricNameError(());
+
+/// A valid metric name. This is the statsd name, i.e. without namespace, type or unit.
+///
+/// Metric names cannot be empty, must begin with a letter and can consist of ASCII alphanumerics,
+/// underscores, slashes and periods.
+#[derive(Debug)]
+pub struct MetricName<'a>(&'a str);
```

Comment:
Not sure if this belongs in `constants.rs`, but `MetricUnit` is also part of this module. Maybe we can move both types to a new file like `relay-common/src/metrics.rs`.